sql/core/src/main/scala/org/apache/spark/sql/execution/python/ExtractPythonUDFs.scala

```diff
@@ -64,11 +64,16 @@ private[spark] object ExtractPythonUDFs extends Rule[LogicalPlan] {
             assert(evaluation != null, "Unable to evaluate PythonUDF.  Missing input attributes.")
 
             // Trim away the new UDF value if it was only used for filtering or something.
-            logical.Project(
-              plan.output,
-              plan.transformExpressions {
-                case p: PythonUDF if p.fastEquals(udf) => evaluation.resultAttribute
-              }.withNewChildren(newChildren))
+            val transformed = plan.transformExpressions {
+              case p: PythonUDF if p.fastEquals(udf) => evaluation.resultAttribute
+            }.withNewChildren(newChildren)
+
+            if (plan.isInstanceOf[Aggregate]) {
+              transformed
+            }
```

Comment:
a style nit: put else on the same line as the previous }

also can you add some comment explaining what's happening